Modules/ITS/src/ITSThresholdCalibrationTask.cxx

```diff
@@ -0,0 +1,456 @@
+      delete hCalibrationRMSChipAverage[iScan][iBarrel];
+    }
+  }
+
+  for (int iBarrel = 0; iBarrel < 3; iBarrel++) {
+
+    delete hCalibrationThrNoiseChipSum[iBarrel];
+    delete hCalibrationThrNoiseChipAverage[iBarrel];
+
+    delete hCalibrationThrNoiseRMSChipSum[iBarrel];
+    delete hCalibrationThrNoiseRMSChipAverage[iBarrel];
+  }
+}
+
+void ITSThresholdCalibrationTask::initialize(o2::framework::InitContext& /*ctx*/)
+{
+
+  ILOG(Info, Support) << "initialize ITSThresholdCalibrationTask" << ENDM;
+
+  mRunNumberPath = mCustomParameters["runNumberPath"];
+
+  createAllHistos();
+  publishHistos();
+}
+
+void ITSThresholdCalibrationTask::startOfActivity(Activity& /*activity*/)
+{
+  ILOG(Info, Support) << "startOfActivity" << ENDM;
+}
+
+void ITSThresholdCalibrationTask::startOfCycle()
+{
+  ILOG(Info, Support) << "startOfCycle" << ENDM;
+}
+
+void ITSThresholdCalibrationTask::monitorData(o2::framework::ProcessingContext& ctx)
+{
+
+  ILOG(Info, Support) << "START DOING QC General" << ENDM;
+  const auto tunString = ctx.inputs().get<gsl::span<char>>("tunestring");
+  const auto runType = ctx.inputs().get<short int>("runtype");
+  const auto scanType = ctx.inputs().get<char>("scantype");
+
+  string inString(tunString.begin(), tunString.end());
+
+  Int_t iScan;
+  Double_t CalibrationValue;
+  if (scanType == 'V')
+    iScan = 0;
+  else if (scanType == 'I')
+    iScan = 1;
+  else if (scanType == 'T')
+    iScan = 2;
+
+  auto splitRes = split(inString, "Stave:");
+
+  for (auto StaveStr : splitRes) {
+    if (StaveStr.size() > 0) {
+      CalibrationResStruct result = CalibrationParser(StaveStr);
+
+      int currentStave = StaveBoundary[result.Layer] + result.Stave + 1;
+      int currentChip = result.ChipID + 1;
+      int iBarrel = getBarrel(result.Layer);
+
+      if (scanType == 'V') {
+        CalibrationValue = result.VCASN;
+      } else if (scanType == 'I') {
+        CalibrationValue = result.ITHR;
+      }
+      if (scanType == 'T') {
+        CalibrationValue = result.THR;
+        hCalibrationThrNoiseChipSum[iBarrel]->SetBinContent(currentChip, currentStave, result.Noise + hCalibrationThrNoiseChipSum[iBarrel]->GetBinContent(currentChip, currentStave));
+        hCalibrationThrNoiseRMSChipSum[iBarrel]->SetBinContent(currentChip, currentStave, result.NoiseRMS + hCalibrationThrNoiseRMSChipSum[iBarrel]->GetBinContent(currentChip, currentStave));
+      }
+
+      hCalibrationChipSum[iScan][iBarrel]->SetBinContent(currentChip, currentStave, CalibrationValue + hCalibrationChipSum[iScan][iBarrel]->GetBinContent(currentChip, currentStave));
+      hCalibrationRMSChipSum[iScan][iBarrel]->SetBinContent(currentChip, currentStave, result.RMS + hCalibrationRMSChipSum[iScan][iBarrel]->GetBinContent(currentChip, currentStave));
+
+      hCalibrationChipCounts[iScan][iBarrel]->SetBinContent(currentChip, currentStave, hCalibrationChipCounts[iScan][iBarrel]->GetBinContent(currentChip, currentStave) + 1);
```

Comment:
This histo is to count how many times a chip is received? In principle every chip is received only once but I think it's a good idea to keep the plot so that in case of issue we can check if a chip is received twice (did not happen so far). Here you can simplify with:
```
histo->Fill(currentChip, currentStave)
```